src/oscore/oscore.c

```diff
@@ -0,0 +1,463 @@
+/* -*- Mode: C; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+
+/*
+ * Copyright (c) 2018, SICS, RISE AB
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ */
+
+/**
+ * @file oscore.c
+ * @brief An implementation of the Object Security for Constrained RESTful
+ * Enviornments (RFC 8613).
+ *
+ * \author Martin Gunnarsson  <martin.gunnarsson@ri.se>
+ * adapted to libcoap and major rewrite
+ *     Peter van der Stok <consultancy@vanderstok.org>
+ *     on request of Fairhair alliance
+ * adapted for libcoap integration
+ *      Jon Shallow <supjps-libcoap@jpshallow.com>
+ *
+ */
+
+#include "coap3/coap_internal.h"
+#include "stdio.h"
+#include <stdbool.h>
+
+#define AAD_BUF_LEN 60 /* length of aad_buffer */
+#define MAX_IV_LEN  10 /* maximum length of iv buffer */
```

Comment:
This repeats a `#define`in another source file. I suggest moving this into the oscore_internal header for consistency. (The same may be true for `AAD_BUF_LEN` which is set to 200 elsewhere.)

Reply:
Deleted both #define as they are no longer relevant.